robocop/checkers/errors.py

```diff
@@ -271,9 +271,7 @@ def visit_If(self, node):  # noqa
         self.parse_errors(node)
         self.generic_visit(node)
 
-    def visit_For(self, node):  # noqa
-        self.parse_errors(node)
-        self.generic_visit(node)
+    visit_For = visit_If
```

Comment:
Now when I look at this, I missed new RF 5.0 syntax here. visit_While, visit_Try. But it's still catched (ie in invalid-if). The question is if we need it at all - the good indicator would be running ``robocop --threshold E -r all`` on Robot Framework atests with this code and without visit_For, visit_If and compare number of reported issues.

Reply:
I've confirmed that we need it and we also need visit_While (we're missing some issues). I will add it & update tests for parsing-error